coreneuron/io/reports/report_handler.cpp

```diff
@@ -155,6 +165,77 @@ VarsToReport ReportHandler::get_compartment_vars_to_report(const NrnThread& nt,
     return vars_to_report;
 }
 
+VarsToReport ReportHandler::get_section_vars_to_report(const NrnThread& nt,
+                                                       const std::set<int>& target,
+                                                       double* report_variable,
+                                                       ReportType type) const {
+    VarsToReport vars_to_report;
+    const auto* mapinfo = static_cast<NrnThreadMappingInfo*>(nt.mapping);
+    if (!mapinfo) {
+        std::cerr << "[COMPARTMENTS] Error : mapping information is missing for a Cell group "
+                  << nt.ncell << '\n';
+        nrn_abort(1);
+    }
+
+    for (int i = 0; i < nt.ncell; i++) {
+        int gid = nt.presyns[i].gid_;
+        if (target.find(gid) != target.end()) {
+            CellMapping* cell_mapping = mapinfo->get_cell_mapping(gid);
+            if (cell_mapping == nullptr) {
+                std::cerr
+                    << "[COMPARTMENTS] Error : Compartment mapping information is missing for gid "
+                    << gid << '\n';
+                nrn_abort(1);
+            }
+            std::vector<VarWithMapping> to_report;
+            to_report.reserve(cell_mapping->size());
+
+            std::string name;
+            switch (type) {
+                case AxonReport:
+                case AxonCompReport:
+                    name = "axon";
+                    break;
+                case DendriteReport:
+                case DendriteCompReport:
+                    name = "dend";
+                    break;
+                case ApicalReport:
+                case ApicalCompReport:
+                    name = "apic";
+                    break;
+                default:
+                    name = "soma";
+            }
```

Comment:
Maybe you can define a function along the enum for that.

Maybe soma should be the default.